modules/matrix_operations/src/matrix_operations.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright 2021 Kiseleva Anastasia
+
+#include<vector>
+#include "include/matrix_operations.h"
+
+Matrix::Matrix(unsigned int row, unsigned int col) {
+    rows = row;
+    cols = col;
+    if ((row != 0) && (col != 0)) {
+        data.resize(row);
+        for (unsigned int i = 0; i < row; i++) {
+            data[i].resize(col);
+            for (unsigned int j = 0; j < col; j++)
+                data[i][j] = 0;
+        }
+    } else {
+        throw "Wrong size";
+    }
+}
+
+Matrix::~Matrix() {}
+
+Matrix::Matrix(const Matrix& tmp) {
+    rows = tmp.rows;
+    cols = tmp.cols;
+    data = tmp.data;
+}
+
+Matrix Matrix::operator*(Matrix A) {
+    Matrix res(rows, A.cols);
+    if (cols == A.rows) {
+        for (unsigned int i = 0; i < rows; i++) {
+            for (unsigned int j = 0; j < A.cols; j++) {
+                double c = 0.0;
+                for (unsigned int k = 0; k < cols; k++) {
+                    c += data[i][k] * A.data[k][j];
+                }
+                res[i][j] = c;
+            }
+        }
+    } else {
+        throw "Wrong size";
+    }
+    return res;
+}
+
+Matrix Matrix::operator* (double tmp) {
+    Matrix res(rows, cols);
+    for (unsigned int i = 0; i < rows; i++) {
+        for (unsigned int j = 0; j < cols; j++) {
+            res[i][j] = data[i][j] * tmp;
+        }
+    }
+    return res;
+}
+
+Matrix Matrix::operator/ (double tmp) {
+    Matrix res(rows, cols);
+    for (unsigned int i = 0; i < rows; i++) {
+        for (unsigned int j = 0; j < cols; j++) {
+            res[i][j] = data[i][j] / tmp;
+        }
+    }
+    return res;
+}
+
+std::vector<double> Matrix::operator*(std::vector<double> tmp) {
+    std::vector<double> res(rows);
+    if (tmp.size() == cols) {
+        for (unsigned int i = 0; i < rows; i++) {
+           double c = 0.0;
+            for (unsigned int j = 0; j < cols; j++) {
+                c += data[i][j] * tmp[j];
+            }
+            res[i] = c;
+        }
+    } else {
+        throw "Wrong size";
+    }
+    return res;
+}
+
+Matrix Matrix::operator+(Matrix A) {
+    Matrix res(rows, cols);
+    if ((rows == A.rows) && (cols == A.cols)) {
+        for (unsigned int i = 0; i < rows; i++) {
+            for (unsigned int j = 0; j < cols; j++) {
+                res[i][j] = data[i][j] + A.data[i][j];
+            }
+        }
+    } else {
+        throw "Wrong size";
+    }
+    return res;
+}
+
+Matrix Matrix::operator-(Matrix A) {
+    Matrix res(rows, cols);
+    if ((rows == A.rows) && (cols == A.cols)) {
+        for (unsigned int i = 0; i < rows; i++) {
+            for (unsigned int j = 0; j < cols; j++) {
+                res[i][j] = data[i][j] - A.data[i][j];
+            }
+        }
+    } else {
+        throw "Wrong size";
+    }
+    return res;
+}
+
+Matrix* Matrix::operator=(Matrix A) {
+    rows = A.rows;
+    cols = A.cols;
+    data = A.data;
+    return this;
+}
```

Comment:
Оператор присваивания реализован неверно. Он должен возвращать ссылку на объект класса, а не указатель. Также нужно сделать проверку на самоприсваивание. И снова константная ссылка на аргумент...
```suggestion
Matrix& Matrix::operator=(const Matrix& A) {
    if (this == &A) return *this;
    rows = A.rows;
    cols = A.cols;
    data = A.data;
    return *this;
}
```